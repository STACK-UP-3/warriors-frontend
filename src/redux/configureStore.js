import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
