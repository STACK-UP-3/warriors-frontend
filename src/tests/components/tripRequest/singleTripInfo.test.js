import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import SingleTripInfo from '../../../components/tripRequests/viewTripRequests/singleTripInfo';
import configureStore from '../../../redux/configureStore';
import { Router } from 'react-router-dom';

describe('*************** Testing the single Trip info component With full information ***************',()=>{

    const store = configureStore();
    let data, accommodationData, wrapper;
    const { location } = window;

    beforeEach(()=>{
        delete window.location;
        window.location = { assign: jest.fn(),search:'' };
        data = {
                email: "firaduk@yahoo.com",
                manager: "NDOLI Jack",
                name:'User For Testing',
                
                result: {
                    id: 2, 
                    trip_id: 2, 
                    status: "pending", 
                    tripRequest: [{
                        origin:'origin city',
                        destination:'destination city',
                        dateOfReturn:'2020-06-15',
                        dateOfTravel:'2020-07-15',
                        travelReason: 'managerTest',
                        accommodation_id: 1,
                        }]},
            };
        
        accommodationData={
            data: {
                status: 200, 
                message: 'Accommodation retrieved successfully', 
                data: {
                    cost: '$78',
                    createdAt: '2020-06-15',
                    description: 'A cool Place',
                    id: 1,
                    images: [{
                        details: 'Back view of accommodation',
                        id: 2,
                        url: 'https://res.cloudinary.com/dcna45id5/image/upload/v1586427877/samples/landscapes/nature-mountains.jpg'
                    }],
                    location: 'Kigali',
                    name: 'Panter Hotel',
                    owner: 'Robert',
                    rooms: [{}],
                    services: [{
                        id: 1, 
                        name: 'service1', 
                        description: 'Best Hotel Ever', 
                        cost: '$54'
                    }],
                    status: 'available',
                    updatedAt: '2020-06-15',
                }
                }
        }
        
        wrapper =  mount(
            <Provider store={ store }> 
                <SingleTripInfo information={ data } accommodation={accommodationData} />
            </Provider>
        );
    });
    
    afterAll(() => {
		window.location = location;
    });
    
    it('Should render the SingleTripInfo component correctly', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
    
    it('Should test on click buttons of the SingleTripInfo component correctly', ()=>{
        const historyMock ={ 
            replace: jest.fn(),
            length: 0,
            location: { 
                pathname: '',
                search: '',
                state: '',
                hash: ''
            },
            action: 'REPLACE',
            push: jest.fn(),
            go: jest.fn(),
            goBack: jest.fn(), 
            goForward: jest.fn(),
            block: jest.fn(),
            listen: jest.fn(),
            createHref: jest.fn()
                             } ;
        
        wrapper =  mount(
            <Provider store={ store }> 
                <Router history={historyMock}>
                    <SingleTripInfo information={ data } accommodation={accommodationData} />
                </Router>
            </Provider>
        );
        
        wrapper.find('[className="backToTrips"]').simulate('click');
        wrapper.find('button').simulate('click');
        
        expect(window.location.assign).toHaveBeenCalled();
    });
    
});

describe('*************** Testing the single Trip info  component missing some information and Disabled button  ***************',()=>{

    const store = configureStore();
    let data, accommodationData, wrapper;
    const { location } = window;

    beforeEach(()=>{
        delete window.location;
        window.location = { assign: jest.fn(),search:'' };
        data = {
                email: "firaduk@yahoo.com",
                manager: "NDOLI Jack",
                name:'User For Testing',
                
                result: {
                    id: 2, 
                    trip_id: 2, 
                    status: "accepted", 
                    tripRequest: [{
                        origin:'origin city',
                        destination:'destination city',
                        dateOfReturn:'2020-06-15',
                        dateOfTravel:'2020-07-15',
                        travelReason: 'managerTest',
                        accommodation_id: 1,
                        }]},
            };
        
        accommodationData={
            data: {
                status: 200, 
                message: 'Accommodation retrieved successfully', 
                data: {
                    cost: '$78',
                    createdAt: '2020-06-15',
                    description: null,
                    id: 1,
                    images: [{
                        details: 'Back view of accommodation',
                        id: 2,
                        url: 'https://res.cloudinary.com/dcna45id5/image/upload/v1586427877/samples/landscapes/nature-mountains.jpg'
                    }],
                    location: 'Kigali',
                    name: null,
                    owner: 'Robert',
                    rooms: [{}],
                    services: [{
                        id: 1, 
                        name: null, 
                        description: null, 
                        cost: null
                    }],
                    status: 'available',
                    updatedAt: '2020-06-15',
                }
                }
        }
        
        wrapper =  mount(
            <Provider store={ store }> 
                <SingleTripInfo information={ data } accommodation={accommodationData} />
            </Provider>
        );
    });
    
    afterAll(() => {
		window.location = location;
    });
    
    it('Should render the SingleTripInfo component correctly', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
    
    it('Should test on click buttons of the SingleTripInfo component correctly', ()=>{
        const historyMock ={ 
            replace: jest.fn(),
            length: 0,
            location: { 
                pathname: '',
                search: '',
                state: '',
                hash: ''
            },
            action: 'REPLACE',
            push: jest.fn(),
            go: jest.fn(),
            goBack: jest.fn(),
            goForward: jest.fn(),
            block: jest.fn(),
            listen: jest.fn(),
            createHref: jest.fn()
                             } ;
        
        wrapper =  mount(
            <Provider store={ store }> 
                <Router history={historyMock}>
                    <SingleTripInfo information={ data } accommodation={accommodationData} />
                </Router>
            </Provider>
        );
        
        wrapper.find('[className="backToTrips"]').simulate('click');
        wrapper.find('button').simulate('click');
        
        expect(historyMock.push).toHaveBeenCalled();
    });
    
});