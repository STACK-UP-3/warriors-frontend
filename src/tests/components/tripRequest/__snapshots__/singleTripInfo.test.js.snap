// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the single Trip info  component missing some information and Disabled button  *************** Should render the SingleTripInfo component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <_default
    accommodation={
      Object {
        "data": Object {
          "data": Object {
            "cost": "$78",
            "createdAt": "2020-06-15",
            "description": null,
            "id": 1,
            "images": Array [
              Object {
                "details": "Back view of accommodation",
                "id": 2,
                "url": "https://res.cloudinary.com/dcna45id5/image/upload/v1586427877/samples/landscapes/nature-mountains.jpg",
              },
            ],
            "location": "Kigali",
            "name": null,
            "owner": "Robert",
            "rooms": Array [
              Object {},
            ],
            "services": Array [
              Object {
                "cost": null,
                "description": null,
                "id": 1,
                "name": null,
              },
            ],
            "status": "available",
            "updatedAt": "2020-06-15",
          },
          "message": "Accommodation retrieved successfully",
          "status": 200,
        },
      }
    }
    information={
      Object {
        "email": "firaduk@yahoo.com",
        "manager": "NDOLI Jack",
        "name": "User For Testing",
        "result": Object {
          "id": 2,
          "status": "accepted",
          "tripRequest": Array [
            Object {
              "accommodation_id": 1,
              "dateOfReturn": "2020-06-15",
              "dateOfTravel": "2020-07-15",
              "destination": "destination city",
              "origin": "origin city",
              "travelReason": "managerTest",
            },
          ],
          "trip_id": 2,
        },
      }
    }
  >
    <div
      className="tripInformation"
    >
      <div
        className="header"
      >
        <div
          className="backToTrips"
          onClick={[Function]}
        >
           
          <i
            className="material-icons"
          >
            arrow_back
          </i>
           Back to trips
        </div>
        <button
          className="btn waves-effect waves-light"
          disabled={true}
          onClick={[Function]}
        >
           EDIT TRIP REQUEST 
        </button>
      </div>
      <h5
        className="title"
      >
        Trip Number:
        2
      </h5>
      <div
        className="content"
      >
        <div
          className="infoRow"
        >
          <span>
            Name: 
          </span>
              
          <b>
            User For Testing
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Email: 
          </span>
              
          <b>
            firaduk@yahoo.com
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            From: 
          </span>
              
          <b>
            origin city
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            To:
          </span>
              
          <b>
            destination city
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Date of Travel:
          </span>
              
          <b>
            2020-07-15
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Date of return: 
          </span>
              
          <b>
            2020-06-15
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Manager: 
          </span>
              
          <b>
            NDOLI Jack
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Travel Reason: 
          </span>
              
          <b>
            managerTest
          </b>
        </div>
        <div
          className="accommodation"
        >
          <span
            className="infoRow"
          >
            Accomodation
          </span>
          <div
            className="accommodationRow"
          >
            <div
              className="accommodationImg"
            >
              <img
                alt="hotel picture"
                src="https://res.cloudinary.com/dcna45id5/image/upload/v1586427877/samples/landscapes/nature-mountains.jpg"
              />
            </div>
            <div
              className="accomodationDetails"
            >
              <div
                className="title"
              >
                <b>
                  Hotel Name
                </b>
              </div>
              <div
                className="serviceDetails"
              >
                <b>
                  serviceName
                </b>
                <span>
                  Cost: 
                  $0.0
                </span>
              </div>
              <div
                className="description"
              >
                Simple Description
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </_default>
</Provider>
`;

exports[`*************** Testing the single Trip info component With full information *************** Should render the SingleTripInfo component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <_default
    accommodation={
      Object {
        "data": Object {
          "data": Object {
            "cost": "$78",
            "createdAt": "2020-06-15",
            "description": "A cool Place",
            "id": 1,
            "images": Array [
              Object {
                "details": "Back view of accommodation",
                "id": 2,
                "url": "https://res.cloudinary.com/dcna45id5/image/upload/v1586427877/samples/landscapes/nature-mountains.jpg",
              },
            ],
            "location": "Kigali",
            "name": "Panter Hotel",
            "owner": "Robert",
            "rooms": Array [
              Object {},
            ],
            "services": Array [
              Object {
                "cost": "$54",
                "description": "Best Hotel Ever",
                "id": 1,
                "name": "service1",
              },
            ],
            "status": "available",
            "updatedAt": "2020-06-15",
          },
          "message": "Accommodation retrieved successfully",
          "status": 200,
        },
      }
    }
    information={
      Object {
        "email": "firaduk@yahoo.com",
        "manager": "NDOLI Jack",
        "name": "User For Testing",
        "result": Object {
          "id": 2,
          "status": "pending",
          "tripRequest": Array [
            Object {
              "accommodation_id": 1,
              "dateOfReturn": "2020-06-15",
              "dateOfTravel": "2020-07-15",
              "destination": "destination city",
              "origin": "origin city",
              "travelReason": "managerTest",
            },
          ],
          "trip_id": 2,
        },
      }
    }
  >
    <div
      className="tripInformation"
    >
      <div
        className="header"
      >
        <div
          className="backToTrips"
          onClick={[Function]}
        >
           
          <i
            className="material-icons"
          >
            arrow_back
          </i>
           Back to trips
        </div>
        <button
          className="btn waves-effect waves-light"
          disabled={false}
          onClick={[Function]}
        >
           EDIT TRIP REQUEST 
        </button>
      </div>
      <h5
        className="title"
      >
        Trip Number:
        2
      </h5>
      <div
        className="content"
      >
        <div
          className="infoRow"
        >
          <span>
            Name: 
          </span>
              
          <b>
            User For Testing
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Email: 
          </span>
              
          <b>
            firaduk@yahoo.com
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            From: 
          </span>
              
          <b>
            origin city
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            To:
          </span>
              
          <b>
            destination city
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Date of Travel:
          </span>
              
          <b>
            2020-07-15
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Date of return: 
          </span>
              
          <b>
            2020-06-15
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Manager: 
          </span>
              
          <b>
            NDOLI Jack
          </b>
        </div>
        <div
          className="infoRow"
        >
          <span>
            Travel Reason: 
          </span>
              
          <b>
            managerTest
          </b>
        </div>
        <div
          className="accommodation"
        >
          <span
            className="infoRow"
          >
            Accomodation
          </span>
          <div
            className="accommodationRow"
          >
            <div
              className="accommodationImg"
            >
              <img
                alt="hotel picture"
                src="https://res.cloudinary.com/dcna45id5/image/upload/v1586427877/samples/landscapes/nature-mountains.jpg"
              />
            </div>
            <div
              className="accomodationDetails"
            >
              <div
                className="title"
              >
                <b>
                  Panter Hotel
                </b>
              </div>
              <div
                className="serviceDetails"
              >
                <b>
                  service1
                </b>
                <span>
                  Cost: 
                  $54
                </span>
              </div>
              <div
                className="description"
              >
                Best Hotel Ever
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </_default>
</Provider>
`;
